src/steg86/binary.rs

```diff
@@ -406,15 +443,21 @@ impl Text {
                         ));
                     }
 
-                    code
+                    instruction
                 };
 
+                let code = instruction.code();
                 let tuple = SEMANTIC_PAIRS
                     .iter()
                     .find(|&&t| code == t.0 || code == t.1)
                     .unwrap();
 
-                header_bits.push(code != tuple.0);
+                if Self::is_commutative(&code) {
+                    let (r0, r1) = (instruction.op0_register(), instruction.op1_register());
+                    header_bits.push((r0 as usize) < (r1 as usize));
```

Comment:
Just to make sure I understand: does this correctly handle the case where `r0 == r1`? I believe we skip over those during embedding since there's no information we can add, so I wanted to make sure we're not incorrectly extracting information from them here.

Reply:
It's not handled here (same below) because they never end up in the profile()'s output

Reply:
Whoops, you're right. I've forgotten my own code :slightly_smiling_face: